pkg/apis/crd.projectcalico.org/v1/bgpconfig.go

```diff
@@ -0,0 +1,140 @@
+
+	// ASNumber is the default AS number used by a node. [Default: 64512]
+	ASNumber *numorstring.ASNumber `json:"asNumber,omitempty" validate:"omitempty" confignamev1:"as_num"`
+
+	// ServiceLoadBalancerIPs are the CIDR blocks for Kubernetes Service LoadBalancer IPs.
+	// Kubernetes Service status.LoadBalancer.Ingress IPs will only be advertised if they are within one of these blocks.
+	ServiceLoadBalancerIPs []ServiceLoadBalancerIPBlock `json:"serviceLoadBalancerIPs,omitempty" validate:"omitempty,dive" confignamev1:"svc_loadbalancer_ips"`
+
+	// ServiceExternalIPs are the CIDR blocks for Kubernetes Service External IPs.
+	// Kubernetes Service ExternalIPs will only be advertised if they are within one of these blocks.
+	ServiceExternalIPs []ServiceExternalIPBlock `json:"serviceExternalIPs,omitempty" validate:"omitempty,dive" confignamev1:"svc_external_ips"`
+
+	// ServiceClusterIPs are the CIDR blocks from which service cluster IPs are allocated.
+	// If specified, Calico will advertise these blocks, as well as any cluster IPs within them.
+	ServiceClusterIPs []ServiceClusterIPBlock `json:"serviceClusterIPs,omitempty" validate:"omitempty,dive" confignamev1:"svc_cluster_ips"`
+
+	// Communities is a list of BGP community values and their arbitrary names for tagging routes.
+	Communities []Community `json:"communities,omitempty" validate:"omitempty,dive" confignamev1:"communities"`
+
+	// PrefixAdvertisements contains per-prefix advertisement configuration.
+	PrefixAdvertisements []PrefixAdvertisement `json:"prefixAdvertisements,omitempty" validate:"omitempty,dive" confignamev1:"prefix_advertisements"`
+
+	// ListenPort is the port where BGP protocol should listen. Defaults to 179
+	// +kubebuilder:validation:Minimum:=1
+	// +kubebuilder:validation:Maximum:=65535
+	ListenPort uint16 `json:"listenPort,omitempty" validate:"omitempty,gt=0" confignamev1:"listen_port"`
+
+	// Optional BGP password for full node-to-mesh peerings.
+	// This field can only be set on the default BGPConfiguration instance and requires that NodeMesh is enabled
+	// +optional
+	NodeMeshPassword *BGPPassword `json:"nodeMeshPassword,omitempty" validate:"omitempty" confignamev1:"node_mesh_password"`
+
+	// Time to allow for software restart for node-to-mesh peerings.  When specified, this is configured
+	// as the graceful restart timeout.  When not specified, the BIRD default of 120s is used.
+	// This field can only be set on the default BGPConfiguration instance and requires that NodeMesh is enabled
+	// +optional
+	NodeMeshMaxRestartTime *metav1.Duration `json:"nodeMeshMaxRestartTime,omitempty" confignamev1:"node_mesh_restart_time"`
+
+	// BindMode indicates whether to listen for BGP connections on all addresses (None)
+	// or only on the node's canonical IP address Node.Spec.BGP.IPvXAddress (NodeIP).
+	BindMode string `json:"bindMode,omitempty" validate:"omitempty,oneof=None NodeIP"`
```

Comment:
Strictly speaking, we could remove all of the other fields and just include this one (since this is the only field that this controller cares about). 

Not sure if @tmjd has any preference on that? I don't really mind either way.

Reply:
@tmjd any preference ? Let us know